qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -308,17 +309,79 @@ def handler(x):
                            " in template %d" %
                            (key, self._id, self._md_template.id))
 
-    def __setitem__(self, key, value):
-        r"""Sets the metadata value for the category `key`
+    def __setitem__(self, column, value):
+        r"""Sets the metadata value for the category `column`
 
         Parameters
         ----------
-        key : str
-            The metadata category
-        value : obj
-            The new value for the category
+        column : str
+            The column to update
+        value : str
+            The value to set. This is expected to be a str on the assumption
+            that psycopg2 will cast as necessary when updating.
+
+        Raises
+        ------
+        QiitaDBColumnError
+            If the column does not exist in the table
```

Comment:
Can you add the ValueError raised when value and database column types are not compatible here?